client/asset/dcr/dcr.go

```diff
@@ -158,7 +159,6 @@ var (
 // rpcClient is an rpcclient.Client, or a stub for testing.
 type rpcClient interface {
 	EstimateSmartFee(ctx context.Context, confirmations int64, mode chainjson.EstimateSmartFeeMode) (*chainjson.EstimateSmartFeeResult, error)
```

Comment:
Fee estimation does not work in SPV mode, so I see `[WRN] CORE[dcr]: Unable to get optimal fee rate, using fallback of 20: -9: RPC passthrough requires dcrd RPC synchronization` when making the reg fee transaction.

```
[WRN] CORE[dcr]: Unable to get optimal fee rate, using fallback of 20: -9: RPC passthrough requires dcrd RPC synchronization
[DBG] CORE[dcr]: Change output size = 36, addr = SsoSsZf84s7Zoy2CVJUDVE2V1o6bJ3A94fy
[DBG] CORE[dcr]: 3 signature cycles to converge on fees for tx eb801ec2e893b8bfb1023a3f3055defc6a7995bc357d075fb95b119fa3542f57: min rate = 20, actual fee rate = 20 (5040 for 252 bytes), change = true
```

In `PayFee`, there's no `feeSuggestion` provided to `feeRateWithFallback`, so it has to use the fallback, not ideal.
One solution would be to use `dc.fetchFeeRate` in `Register` and pass that to a modified `PayFee`.

i.e.

```diff
diff --git a/client/core/core.go b/client/core/core.go
index 5d3c61f9..4f020210 100644
--- a/client/core/core.go
+++ b/client/core/core.go
@@ -2542,7 +2542,8 @@ func (c *Core) Register(form *RegisterForm) (*RegisterResult, error) {
        c.log.Infof("Attempting registration fee payment to %s, account ID %v, of %d units of %s. "+
                "Do NOT manually send funds to this address even if this fails.",
                incompleteReg.regRes.Address, acctID, incompleteReg.regRes.Fee, regFeeAssetSymbol)
-       coin, err := wallet.PayFee(incompleteReg.regRes.Address, incompleteReg.regRes.Fee)
+       feeRateSuggestion := dc.fetchFeeRate(feeAsset.ID)
+       coin, err := wallet.PayFee(incompleteReg.regRes.Address, incompleteReg.regRes.Fee, feeRateSuggestion)
        if err != nil {
                return nil, newError(feeSendErr, "error paying registration fee: %v", err)
        }
```

```diff
diff --git a/client/asset/dcr/dcr.go b/client/asset/dcr/dcr.go
index e11fd7b1..d4acca7d 100644
--- a/client/asset/dcr/dcr.go
+++ b/client/asset/dcr/dcr.go
@@ -750,10 +750,16 @@ func (dcr *ExchangeWallet) feeRate(confTarget uint64) (uint64, error) {
 // feeRateWithFallback attempts to get the optimal fee rate in atoms / byte via
 // FeeRate. If that fails, it will return the configured fallback fee rate.
 func (dcr *ExchangeWallet) feeRateWithFallback(confTarget, feeSuggestion uint64) uint64 {
-       feeRate, err := dcr.feeRate(confTarget)
-       if err == nil {
-               dcr.log.Tracef("Obtained local estimate for %d-conf fee rate, %d", confTarget, feeRate)
-               return feeRate
+       var err error
+       if !dcr.spvMode {
+               var feeRate uint64
+               feeRate, err = dcr.feeRate(confTarget)
+               if err == nil {
+                       dcr.log.Tracef("Obtained local estimate for %d-conf fee rate, %d", confTarget, feeRate)
+                       return feeRate
+               }
+       } else {
+               err = errors.New("SPV does not support estimatesmartfee")
        }
        if feeSuggestion > 0 && feeSuggestion < dcr.fallbackFeeRate && feeSuggestion < dcr.feeRateLimit {
```

With the above, I then see this:

```
[TRC] CORE[dcr]: feeRateWithFallback using caller's suggestion for 1-conf fee rate, 10. Local estimate unavailable ("SPV does not support estimatesmartfee")
[DBG] CORE[dcr]: Change output size = 36, addr = SsUzFsEy1kG75JhPfmgkGwdouw9Nh4VArVr
[DBG] CORE[dcr]: 2 signature cycles to converge on fees for tx fc465eecb5699ed9d93fd69f377c3736f0c29e48f207ea461d72b295bea9a40d: min rate = 10, actual fee rate = 10 (2520 for 252 bytes), change = true
```

The RegisterResult could be modified to include a suggested rate for the asset.  Also, the regFees part of the config response could also be modified to include a current suggested fee rate, but I don't like that idea for several reasons.  I think fetching the fee rate when needed is preferable.

Reply:
@buck54321 Related to the above comment, I noticed that `(*Core).feeSuggestion` has an unused `isBase bool` input arg.  Was this a leftover, or part of future work plans?